acinclude/squid-util.m4

```diff
@@ -214,6 +214,30 @@ AC_DEFUN([SQUID_YESNO],[
   AS_IF([test "$1" != "yes" -a "$1" != "no"],[AC_MSG_ERROR([Bad argument for $2: "$1". Expecting "yes", "no", or no argument.])])
 ])
 
+dnl Check that a library is actually available, useable,
+dnl and where its pieces are (eg headers and hack macros)
+dnl Parameters for this macro are:
+dnl 1) library name (without 'lib' prefix)
+dnl 2) necessary library checks (to be executed by this macro unless the use of the library is disabled)
+dnl   These checks should set LIBFOO_LIBS variable for automake
+dnl   and ensure that it is empty on detection/validation failures.
+AC_DEFUN([SQUID_CHECK_LIB_WORKS],[
+AS_IF([m4_translit([test "x$with_$1" != "xno"], [-+.], [___])],[
+  $2
+  AS_IF([! test -z m4_toupper(m4_translit(["$LIB$1_LIBS"], [-+.], [___]))],[
+    m4_toupper(m4_translit([CXXFLAGS="$LIB$1_CFLAGS $CXXFLAGS"], [-+.], [___]))
```

Comment:
Squid no longer builds any binaries with C compiler, only individual objects or the lib/ directory libraries. So CFLAGS is not useful for any third-party libraries linked to our final binaries.

CPPFLAGS is preprocessor-only flags - which *could* work, but is technically wrong since the C++ compiler may also need these flags and passes them as-needed to the preprocessor. So CXXFLAGS is better.

(well, to be complete/pedantic AM_CXXFLAGS is the one we should be setting, but that requires wider changes I left out of scope for now).

Reply:
Assume that I know all of the things you have pointed out when I say that CXXFLAGS is the one we should be updating and not the others.

Reply:
> Assume that I know all of the things you have pointed out

Glad we are now on the same page regarding those facts and considerations!

> CXXFLAGS is the one we should be updating and not the others.

Why? I see no connection between the discussion so far and the above assertion.

AFAICT, the earlier [attempt](https://github.com/squid-cache/squid/pull/1333/files#r1182484984) to defend PR changes contradicted the [facts and considerations](https://github.com/squid-cache/squid/pull/1333/files#r1182575879) that your recent [comment](https://github.com/squid-cache/squid/pull/1333/files#r1215711568) concurs with. Those facts and considerations imply that several variables should be updated (and that CXXFLAGS variable should be updated in a different way). To address this change request, the PR should either make the corresponding changes or invalidate those suggestions. Neither has happened.

Reply:
This should be resolved by commit [d631d08](https://github.com/squid-cache/squid/pull/1333/commits/d631d084447b78440d4a71942eb34c1e739ddf5d)

Refernce: https://autotools.info/pkgconfig/pkg_check_modules.html#idm45459109695696